[HACK week][Swift6 compatibility]Exploration and early updates

iamgabrielma opened the pull request:
## Description
Task defined on: WOOMOB-1001

Upgrades `SWIFT_STRICT_CONCURRENCY` checks from minimal to targeted only for `DEBUG` builds. Release and Release-alpha builds remain the same.

- Increases the warnings from 16 to 120+ due new checks, but compiles.
- Testing done mostly around POS
  - Concurrency minimal: Checks only inside concurrency features
    - Code directly inside actors, async functions, and `MainActor`
    - Treats the boundary between concurrent and non-concurrent code loosely
    - Rest is completely ignored
  - Concurrency targeted: 
    - Once a file/module contains any concurrency feature, the entire file/module is checked
    - Enforces safety at the boundaries between concurrent and non-concurrent code
    - Rest is completely ignored

## Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

iamgabrielma committed:
deprecated unlabeled trailing closure

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
make funcs async

iamgabrielma committed:
wrap async await in tasks

iamgabrielma committed:
clear unnecessary await

iamgabrielma committed:
lint

iamgabrielma committed:
mark nonisolated

immutable and set during init, no state mutation during runtime, should be safe to mark as nonisolated

iamgabrielma committed:
make navigation nonisolated

iamgabrielma committed:
fix lint on temporary comment

iamgabrielma committed:
remove unnecessary task block

iamgabrielma committed:
tests for POSProductSearchableTests

iamgabrielma committed:
mark preconcurrency imports and notes

iamgabrielma committed:
remove arning noise from unused props

iamgabrielma committed:
raise a notch strict concurrency in debug builds. keep it minimal on release builds

iamgabrielma committed:
revert strict concurrency back to minimal to test failing tests

iamgabrielma committed:
address clearSearchOrders test failure

iamgabrielma committed:
restore reference to VM so is not deallocated immediately when performing the test

iamgabrielma committed:
restore unused value

iamgabrielma committed:
revert removal of unassigned vars so are not deallocated before the test completes

iamgabrielma committed:
restore strict concurrency targeted in debug

iamgabrielma committed:
lint
